### PR TITLE
Compose article

### PR DESCRIPTION
**Pull request by marcussfu**
1.  Add ComposeArticleFlow and follow coordinator
2. POST /api/board/{bid}/article
3. Adjust "getBoardArticles" and "getArticle" from .legacy to .go_pttbbs

### CODE REVIEW

**Review by denkeni — CHANGES_REQUESTED** [review]

File: Ptt/Flows/ComposeArticleFlow/Controllers/ComposeArticleViewController.swift [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    var classText = ""
+    var noticeText = ""
+    let placeholderText = "請輸入文章內容"
+    var boardName = ""
+    
+    lazy var classSelectButton: UIButton = {
+        var classSelectButton = UIButton()
+        classSelectButton.translatesAutoresizingMaskIntoConstraints = false
+        classSelectButton.setTitle("分類選擇", for: .normal)
+        classSelectButton.setTitleColor(UIColor(red:240/255, green:240/255, blue:247/255, alpha:1.0), for: .normal)
```
[the remark this answers is not in the document]
Comment (marcussfu) [reply Ptt/Flows/ComposeArticleFlow/Controllers/ComposeArticleViewController.swift:25]: corrected

File: Ptt/Flows/ComposeArticleFlow/Controllers/ComposeArticleViewController.swift [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    lazy var contentView: UITextView = {
+        let contentView = UITextView()
+        contentView.font = UIFont.systemFont(ofSize: 18)
+        contentView.delegate = self
+        contentView.translatesAutoresizingMaskIntoConstraints = false
+        contentView.textAlignment = .left
+        contentView.backgroundColor = .black
+        contentView.text = placeholderText
+        contentView.textColor = UIColor(red:56/255, green:56/255, blue:61/255, alpha:1.0)
+        contentView.backgroundColor = UIColor(named: "blackColor-23-23-23")
```
[the remark this answers is not in the document]
Comment (marcussfu) [reply Ptt/Flows/ComposeArticleFlow/Controllers/ComposeArticleViewController.swift:51]: corrected.

File: Ptt/Flows/BoardFlow/Controllers/BoardViewController.swift [hunk 3]
```diff
@@ -83,16 +85,14 @@ final class BoardViewController: ASDKViewController<ASDisplayNode>, FullscreenSw
...
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
-
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
         if let selectedRow = tableView.indexPathForSelectedRow {
             tableView.deselectRow(at: selectedRow, animated: true)
         }
+        refresh()
```
[the remark this answers is not in the document]
Comment (marcussfu) [reply Ptt/Flows/BoardFlow/Controllers/BoardViewController.swift:95]: refresh after creative compose

File: Ptt/API/Models/Article.swift [hunk 4]
```diff
@@ -31,9 +31,9 @@ extension Article {
     var titleWithoutCategory : String {
         if let leftBracket = title.firstIndex(of: "["), let rightBracket = title.firstIndex(of: "]") {
             var _title = title
-            let nextRightBracket = _title.index(after: rightBracket)
-            let range = leftBracket...nextRightBracket
-            _title.removeSubrange(range)
+//            let nextRightBracket = _title.index(after: rightBracket)
+//            let range = leftBracket...nextRightBracket
+//            _title.removeSubrange(range)
```
[the remark this answers is not in the document]
Comment (marcussfu) [reply Ptt/API/Models/Article.swift:36]: 所以上正式環境就不會有[測試]了嗎

[the remark this answers is not in the document]
Comment (denkeni) [reply Ptt/API/Models/Article.swift:36]: 應該是，要確認可以去群組問。

[the remark this answers is not in the document]
Comment (marcussfu) [reply Ptt/API/Models/Article.swift:36]: 已確認，只在測試環境有

File: Ptt/Flows/BoardFlow/Controllers/BoardViewController.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcussfu) [reply Ptt/Flows/BoardFlow/Controllers/BoardViewController.swift]: corrected.

File: Ptt/Flows/BoardFlow/BoardCoordinator.swift [hunk 5]
```diff
@@ -45,4 +49,9 @@ private extension BoardCoordinator {
         let articleView = factory.makeArticleView(withBoardArticle: boardArticle)
         router.push(articleView)
     }
+    
+    func showComposeArticleView(withBoardName boardName: String) {
+        let composeArticleView = factory.makeComposeArticleView(withBoardName: boardName)
+        router.push(composeArticleView)
```
[the remark this answers is not in the document]
Comment (marcussfu) [reply Ptt/Flows/BoardFlow/BoardCoordinator.swift]: 用denkeni的方法"把 ComposeArticleViewController 放進一個新的 UINavigationController，再 present 這個 nav "，已經可以顯示navigation bar了，離開ComposeArticleViewController改成dismiss

File: Ptt/Flows/ComposeArticleFlow/Controllers/ComposeArticleViewController.swift [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        initView()
+        initToolBar()
+        initKeyboardToolBar()
+        setConstraint()
```
[the remark this answers is not in the document]
Comment (denkeni) [reply Ptt/Flows/ComposeArticleFlow/Controllers/ComposeArticleViewController.swift:228]: I'll help do it later.

**Review by denkeni — APPROVED** [review]

**Review by denkeni — APPROVED** [review]

**Pull request merged**